pkg/migration/cutover.go

```diff
@@ -168,6 +168,7 @@ func (c *CutOver) algorithmGhost(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+	defer trx.Rollback() //nolint: errcheck
```

Comment:
This is a leak I caught.

Reply:
Shouldn't you rollback only incase of error?

Reply:
I will add a comment to make this clearer.

There is no `trx.Close()` (go idiomatic) you have to chose to commit or rollback. This connection is used for `RENAME` which is DDL, and is non-transactional. So either commit or rollback has the same effect. I chose rollback.